Sources/Fluent/Fluent+History.swift

```diff
@@ -0,0 +1,80 @@
+import Vapor
+import FluentKit
+
+struct RequestQueryHistory: StorageKey {
+    typealias Value = QueryHistory
+}
+
+struct FluentHistoryKey: StorageKey {
+    typealias Value = FluentHistory
+}
+
+struct FluentHistory {
+    let enabled: Bool
+}
+
+extension Application.Fluent.History {
+    var historyEnabled: Bool {
+        return (self.fluent.application.storage[FluentHistoryKey.self]?.enabled) ?? false
+    }
+
+    var history: QueryHistory {
+        guard historyEnabled else { return .init() }
+        return self.fluent.application.storage[RequestQueryHistory.self] ?? .init()
+    }
+
+    /// The queries stored in this lifecycle history
+    public var queries: [DatabaseQuery] {
+        return history.queries
+    }
+
+    /// Start recording the query history
+    public func start() {
+        self.fluent.application.storage[FluentHistoryKey.self] = .init(enabled: true)
+        self.fluent.application.storage[RequestQueryHistory.self] = .init()
+    }
+
+    /// Stop recording the query history
+    public func stop() {
+        self.fluent.application.storage[FluentHistoryKey.self] = .init(enabled: false)
```

Comment:
Nit: There's a lot of duplication of `self.fluent.application.storage[...]`. A pattern I've used elsewhere is to make use of a local `storage` variable. You can see an example of that here: https://github.com/vapor/vapor/blob/master/Sources/Vapor/Sessions/Application%2BSessions.swift. You can see `self.application.storage` is only referenced twice in file. Here, too https://github.com/vapor/jwt/blob/master/Sources/JWT/JWT%2BApple.swift